parachain/runtime/interlay/src/lib.rs

```diff
@@ -1508,6 +1510,46 @@ impl_runtime_apis! {
             let balance = BalanceWrapper::<Balance> { amount };
             Ok(balance)
         }
+
+        fn estimate_escrow_reward_rate(
+            account_id: AccountId,
+            amount: Option<Balance>,
+            lock_time: Option<BlockNumber>,
+            period: BlockNumber,
+        ) -> Result<UnsignedFixedPoint, DispatchError> {
+            // withdraw previous rewards
+            <EscrowRewards as reward::RewardsApi<(), AccountId, Balance>>::withdraw_reward(&(), &account_id, NATIVE_CURRENCY_ID)?;
+            // increase amount and/or lock_time
+            Escrow::round_height_and_deposit_for(&account_id, amount.unwrap_or_default(), lock_time.unwrap_or_default())?;
+            // distribute rewards accrued over block count
+            let reward = EscrowAnnuity::min_reward_per_block().saturating_mul(period.into());
+            <EscrowRewards as reward::RewardsApi<(), AccountId, Balance>>::distribute_reward(&(), NATIVE_CURRENCY_ID, reward)?;
+            let received = <EscrowRewards as reward::RewardsApi<(), AccountId, Balance>>::compute_reward(&(), &account_id, NATIVE_CURRENCY_ID)?;
+            // NOTE: total_locked is same currency as rewards
+            let total_locked = Escrow::locked_balance(&account_id).amount;
+            // rate is received / total_locked
+            Ok(UnsignedFixedPoint::checked_from_rational(received, total_locked).unwrap_or_default())
+        }
+
+        fn estimate_vault_reward_rate(
+            vault_id: VaultId,
+            period: BlockNumber,
+        ) -> Result<UnsignedFixedPoint, DispatchError> {
+            // distribute and withdraw previous rewards
+            Fee::distribute_vault_rewards(&vault_id, NATIVE_CURRENCY_ID)?;
```

Comment:
Is calling `distribute_vault_rewards(...)` required because of the capacity model? There is no initial reward distribution in `estimate_escrow_reward_rate`

Reply:
Yes because there are three layers of vault reward pools whereas there is only one reward pool for escrow